src/shared/blocks/blocks/DistantRadar.ts

```diff
@@ -0,0 +1,318 @@
+					} else {
+						item.radarChunk.Transparency = 1;
+					}
+
+					cloneRadarChunk.Touched.Connect((otherPart) => {
+						if (item.isAvalaible || item.isDetected) {
+							return;
+						}
+
+						if (otherPart.HasTag(TagUtils.allTags.SPECIAL_RADARVIEW)) return; // ignore special tags
+						if (!inputValues.detectSelf && ownDetectablesSet.has(otherPart)) return; // ignore own blocks if this option is set
+
+						// save data
+						item.isDetected = true;
+						item.detectedCords = otherPart.Position;
+					});
+
+					radarChunksList.push(item);
+				}
+
+				isFirstTick = false;
+			}
+
+			if (tickCounter >= inputValues.chunkSpawnRate) {
+				// find and mark new chunk
+				for (let i = 0; i < inputValues.radarChunksCap; i++) {
+					if (radarChunksList[i].isAvalaible) {
+						radarChunksList[i].radarChunk.CFrame = metalBase.CFrame;
+						radarChunksList[i].radarChunk.Position = metalBase.Position;
+						radarChunksList[i].radarChunk.CFrame.mul(new CFrame(inputValues.start, 0, 0));
+						radarChunksList[i].c = 0;
+						radarChunksList[i].isAvalaible = false;
+						radarChunksList[i].isDetected = false;
+
+						break;
+					}
+				}
+
+				tickCounter = 1; // drop tick counter when chunk was selected
+			}
+
+			if (detectQueryCap.size() > 0) {
+				this.output.gps.set("vector3", detectQueryCap[0].add(offset));
+				detectQueryCap.remove(0);
+				this.output.isDetected.set("bool", true);
+			} else {
+				this.output.isDetected.set("bool", false);
+			}
+
+			tickCounter++;
+		});
+
+		this.event.subscribe<[number]>(RunService.Heartbeat as ReadonlyArgsSignal<[number]>, (dt) => {
+			// we run through the created chunks to move them forward
+			for (let i = 0; i < inputValues.radarChunksCap; i++) {
+				if (!radarChunksList[i].isAvalaible) {
+					let isDrop = false;
+
+					// void checking
+					if (radarChunksList[i].radarChunk.Position.Y < -offset.Y) {
+						isDrop = true;
+					}
+
+					// border checking
+					if (radarChunksList[i].c * 100 > inputValues.end) {
+						isDrop = true;
+					}
+
+					// detection checking
+					if (radarChunksList[i].isDetected) {
+						if (detectQueryCap.size() < inputValues.detectQueryCap) {
+							detectQueryCap.push(radarChunksList[i].detectedCords);
+						}
+
+						isDrop = true;
+					}
+
+					if (isDrop) {
+						radarChunksList[i].radarChunk.Position = offset;
+						radarChunksList[i].c = 0;
+						radarChunksList[i].isDetected = false;
+						radarChunksList[i].detectedCords = new Vector3(0, 0, 0);
+						radarChunksList[i].isAvalaible = true;
+					} else {
+						radarChunksList[i].radarChunk.CFrame = radarChunksList[i].radarChunk.CFrame.add(
+							radarChunksList[i].radarChunk.CFrame.RightVector.mul(100),
+						);
+
+						const newSize = (inputValues.chunkEndSize * radarChunksList[i].c * 100) / inputValues.end;
+						radarChunksList[i].radarChunk.Size = new Vector3(100, newSize, newSize);
+
+						radarChunksList[i].c++;
+					}
```

Comment:
if(...){
// code;
return;
}

// else condition

Allows to get rid of extra nestings
I personally hate nestings

```suggestion
					if (isDrop) {
						radarChunksList[i].radarChunk.Position = offset;
						radarChunksList[i].c = 0;
						radarChunksList[i].isDetected = false;
						radarChunksList[i].detectedCords = new Vector3(0, 0, 0);
						radarChunksList[i].isAvalaible = true;
						return;
					}

					radarChunksList[i].radarChunk.CFrame = radarChunksList[i].radarChunk.CFrame.add(
						radarChunksList[i].radarChunk.CFrame.RightVector.mul(100),
					);

					const newSize = (inputValues.chunkEndSize * radarChunksList[i].c * 100) / inputValues.end;
					radarChunksList[i].radarChunk.Size = new Vector3(100, newSize, newSize);

					radarChunksList[i].c++;
```